src/TorchSharp/Utils/TensorAccessor.cs

```diff
@@ -47,6 +48,16 @@ public T[] ToArray()
             if (_tensor.ndim < 2)
                 return (T[])ToNDArray();
 
+            if (_tensor.is_contiguous()) {
+                //This is very fast. And work VERY WELL
+                var shps = _tensor.shape;
+                long TempCount = 1;
+                for (int i = 0; i < shps.Length; i++)
+                    TempCount *= shps[i]; //Theorically the numel is simple as product of each element shape
```

Comment:
Why can't you just use the numel() function here?

Reply:
If i don't remember incorrectly the Numel was Slow, but maybe the current version is fast. No problem i will change that.

Reply:
The implementation is in native code, so I would imagine that libtorch does what it can to make it fast:

```C#
NativeMethods.THSTensor_numel(Handle);
```

Reply:
If there's a performance issue, I guess it could be memoized.